crates/ruff_python_parser/src/lexer/fstring.rs

```diff
@@ -35,49 +17,45 @@ pub(crate) struct FStringContext {
 }
 
 impl FStringContext {
-    pub(crate) const fn new(flags: FStringContextFlags, nesting: u32) -> Self {
+    pub(crate) const fn new(kind: StringKind, nesting: u32) -> Self {
         Self {
-            flags,
+            kind,
```

Comment:
Can we add a `debug_assert` here as a sanity check to make sure that the `kind` is only a f-string?
```rs
    pub(crate) const fn new(kind: StringKind, nesting: u32) -> Self {
		debug_assert!(kind.is_f_string());
		// ...
	}
```